ISO_Fortran_binding.h missing from container package

I'm trying to set up CI using the [LFortran container package](https://github.com/lfortran/lfortran/pkgs/container/lfortran), where lfortran is installed in the prefix `/app`.

Unfortunately it appears the `ISO_Fortran_binding.h` file required by Fortran 2018+ is not installed _anywhere_ in the container:

```bash
$ docker run --rm -it ghcr.io/lfortran/lfortran:latest
Unable to find image 'ghcr.io/lfortran/lfortran:latest' locally
latest: Pulling from lfortran/lfortran
de47083ed7d7: Already exists 
6d420c8e7bd6: Pull complete 
db2f488f1f08: Pull complete 
Digest: sha256:23f07e61a8b79983ae137d18fdc584949fedd4643fb292da7e97c04bdec9c683
Status: Downloaded newer image for ghcr.io/lfortran/lfortran:latest

root@88b18d63581d:/# ls -R /app
/app:
bin  include  share

/app/bin:
lfortran

/app/include:
lfortran

/app/include/lfortran:
impure

/app/include/lfortran/impure:
lfortran_intrinsics.h

/app/share:
lfortran

/app/share/lfortran:
dat_convert.py  dwarf_convert.py  lib

/app/share/lfortran/lib:
lfortran_intrinsic_custom.mod           liblfortran_runtime.so.0
lfortran_intrinsic_ieee_arithmetic.mod  liblfortran_runtime.so.0.62.0
lfortran_intrinsic_iso_c_binding.mod    liblfortran_runtime_static.a
lfortran_intrinsic_iso_fortran_env.mod  omp_lib.mod
liblfortran_runtime.so

root@88b18d63581d:/# find / -name ISO_Fortran_binding.h -print

root@88b18d63581d:/# which clang
/usr/bin/clang
root@88b18d63581d:/# clang --version
Ubuntu clang version 14.0.0-1ubuntu1.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

root@88b18d63581d:/# cat > test.c
#include "ISO_Fortran_binding.h"
root@88b18d63581d:/# clang -E test.c
# 1 "test.c"
# 1 "<built-in>" 1
# 1 "<built-in>" 3
# 361 "<built-in>" 3
# 1 "<command line>" 1
# 1 "<built-in>" 2
# 1 "test.c" 2
test.c:1:10: fatal error: 'ISO_Fortran_binding.h' file not found
#include "ISO_Fortran_binding.h"
         ^~~~~~~~~~~~~~~~~~~~~~~

1 error generated.
root@88b18d63581d:/#
```

Please add `ISO_Fortran_binding.h` to the container package.

Ideally it would go somewhere in the system default include path (e.g. `/usr/local/include` or `/usr/include/x86_64-linux-gnu`) to avoid the need to mess around with include search paths, but `/app/include` would also be reasonable (especially if the container is arranged to inject the directory into `$CPATH`).

--- a/src/bin/lfortran.cpp
+++ b/src/bin/lfortran.cpp
@@ -2563,6 +2563,11 @@ int main_app(int argc, char *argv[]) {
 #endif
     }
 
+    if (opts.print_c_include_dir) {
+        std::cout << LCompilers::LFortran::get_c_include_dir() << std::endl;
+        return 0;
+    }
+
     if(opts.static_link && opts.shared_link) {
         std::cerr << "Options '--static' and '--shared' cannot be used together" << std::endl;
         return 1;

--- a/src/bin/lfortran_command_line_parser.cpp
+++ b/src/bin/lfortran_command_line_parser.cpp
@@ -330,6 +330,7 @@ namespace LCompilers::CommandLineInterface {
         app.add_flag("--linker-path", opts.linker_path, "Use the linker from this path")->capture_default_str()->group(group_backend_codegen_options);
         app.add_option("--target", compiler_options.target, "Generate code for the given target")->capture_default_str()->group(group_backend_codegen_options);
         app.add_flag("--print-targets", opts.print_targets, "Print the registered targets")->group(group_backend_codegen_options);
+        app.add_flag("--print-c-include-dir", opts.print_c_include_dir, "Print the directory containing ISO_Fortran_binding.h")->group(group_backend_codegen_options);
         app.add_flag("--wasm-html", compiler_options.wasm_html, "Generate HTML file using emscripten for LLVM->WASM")->group(group_backend_codegen_options);
         app.add_option("--emcc-embed", compiler_options.emcc_embed, "Embed a given file/directory using emscripten for LLVM->WASM")->group(group_backend_codegen_options);
         app.add_flag("--mlir-gpu-offloading", compiler_options.po.enable_gpu_offloading, "Enables gpu offloading using MLIR backend")->group(group_backend_codegen_options);

--- a/src/bin/lfortran_command_line_parser.h
+++ b/src/bin/lfortran_command_line_parser.h
@@ -59,6 +59,7 @@ namespace LCompilers::CommandLineInterface {
         std::string linker{""};
         std::string linker_path{""};
         bool print_targets = false;
+        bool print_c_include_dir = false;
         bool fixed_form_infer = false;
         bool cpp = false;
         bool cpp_infer = false;

--- a/src/lfortran/utils.cpp
+++ b/src/lfortran/utils.cpp
@@ -108,6 +108,21 @@ std::string get_runtime_library_c_header_dir()
     }
 }
 
+std::string get_c_include_dir()
+{
+    switch (execution_mode)
+    {
+        case ExecutionMode::LFortranDevelopment:
+            return lfortran_exec_path_dir + "/../libasr/runtime";
+        case ExecutionMode::LFortranCtest:
+            return lfortran_exec_path_dir + "/../../libasr/runtime";
+        case ExecutionMode::LFortranInstalled:
+            return lfortran_exec_path_dir + "/" + CMAKE_INSTALL_INCLUDEDIR_RELATIVE + "/lfortran";
+        default:
+            return "";
+    }
+}
+
 std::string get_dwarf_scripts_dir()
 {
     char *env_p = std::getenv("LFORTRAN_DWARF_SCRIPTS_DIR");

--- a/src/lfortran/utils.h
+++ b/src/lfortran/utils.h
@@ -17,6 +17,7 @@ void set_exec_path_and_mode(std::string &executable_path, int &dirname_length);
 std::string get_runtime_library_dir();
 std::string get_runtime_library_header_dir();
 std::string get_runtime_library_c_header_dir();
+std::string get_c_include_dir();
 std::string get_dwarf_scripts_dir();
 int32_t get_exit_status(int32_t err);
 std::string get_kokkos_includedir();

--- a/src/runtime/legacy/CMakeLists.txt
+++ b/src/runtime/legacy/CMakeLists.txt
@@ -82,3 +82,8 @@ install(
     FILES ../../../src/libasr/runtime/lfortran_intrinsics.h
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/lfortran/impure
 )
+
+install(
+    FILES ../../../src/libasr/runtime/ISO_Fortran_binding.h
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/lfortran
+)

